README.md

```diff
@@ -1 +1,244 @@
+* **Phase 4**: Set up a CI/CD pipeline to automate testing, and deployment for `ht_search` project.
+* **Phase 5**: Repeat the process for other projects in the monorepo, ensuring that each project is properly structured and tested.
+* **Phase 6**: Install the monorepo in editable mode, allowing for real-time updates during development.
+  * Define dependencies in editable mode add `develop = true ==> {ht-utils = {path = "../../base/ht_utils", develop = true}`
+  * This approach is useful to develop using the docker image, as it allows you to edit the code in the 
+  monorepo and see the changes reflected in the Docker container without having to rebuild the image every time.
+  * I don't know how to do this yet, but I will figure it out.
+
+
+## Project Set Up
+
+We use `Makefile` and `Dockerfile` to manage the environment set up and build the image simulating equivalent paths 
+in the Docker image and locally.
+
+The deployment process is done through Docker image as showed below:
+
+```
+build:
+    # Copy project files
+	cp -r src $(TMP_DIR)/src
+
+	cp Dockerfile .dockerignore $(TMP_DIR)
+	cp pyproject.toml poetry.lock README.md $(TMP_DIR)
+
+	# Copy the shared package written in pyproject.toml
+	cp -r ../../base/ht_utils $(TMP_DIR)/ht_utils
+	cp -r ../../base/ht_search $(TMP_DIR)/ht_search
+
+	docker build -t $(IMAGE_NAME) $(TMP_DIR)
+
+	rm -rf $(TMP_DIR)
+```
+
+In the `Makefile`, a temporary directory is created with the project and its dependencies to build the local package.
+Then, in Dockerfile, we copy the project files maintaining the same structure as in the temporary directory. Then, we 
+use a multistage Dockerfile to build the image. The first stage is used to install the dependencies, and the second 
+stage is used to copy the code into the image.
+
+**Steps to add a dependency**:
+
+In the Dockerfile,
+* Copy the dependency on the base stage of the Docker image. `COPY ./ht_search /base/ht_search/`
+* Copy the dependency on the final stage of the Docker image. `COPY --chown=${UID}:${GID} ht_search/ ht_search/`
+* In the pyproject.toml file, add the dependency to the [tool.poetry.dependencies] section.
+       ```ht-search = {path = "../../base/ht_search"}```
+
+## Design 
+
+The design of the `index_search_monorepo` is structured to ensure uniformity between projects and to avoid duplicated 
+code. 
+
+1. Monorepo Structure
+The monorepo is organized into two main directories: `base` and `projects`. 
+
+* `base`: Contains shared libraries and utilities that are reused across multiple projects.
+* `projects`: Contains independent projects, each with its own functionality and dependencies.
+
+2. Shared Libraries
+`Shared libraries` are placed in the base directory (e.g., ht_utils and ht_search).
+Each shared library has its own `pyproject.toml` file for dependency management.
+These libraries are installed in non-editable mode, that does not allow real-time updates during development. Right now,
```

Comment:
This will likely end up feeling rather limiting. We will likely want to have a follow-up task to enable better co-development of libraries and applications (which is maybe a key benefit of putting them in a monorepo). This could be something we look at together or with @Ronster2018.